network/net.go

```diff
@@ -419,11 +419,14 @@ func (c *TCPConn) Receive(ctx context.Context) (nm Packet, e error) {
 			log.Error("Couldn't write to buffer:", err)
 		}
 		read += Size(n)
+		b = b[n:]
```

Comment:
👍 👍 👍  for finding and solving this one!
Things like this should be covered by unit tests (you already opened an issue for that?)